it/src/main/java/org/corfudb/universe/node/server/docker/DockerCorfuServer.java

```diff
@@ -0,0 +1,403 @@
+    public void restart() {
+        log.info("Restarting the corfu server: {}", params.getName());
+
+        try {
+            ContainerInfo container = docker.inspectContainer(params.getName());
+            if (container.state().running() || container.state().paused()) {
+                log.warn("The container `{}` already running, should stop before restart", container.name());
+                return;
+            }
+            docker.restartContainer(params.getName());
+        } catch (DockerException | InterruptedException ex) {
+            throw new NodeException("Can't restart container " + params.getName(), ex);
+        }
+    }
+
+    /**
+     * Reconnect a {@link Node} to the network
+     *
+     * @throws NodeException this exception will be thrown if the node can not be reconnected
+     */
+    @Override
+    public void reconnect() {
+        log.info("Reconnecting the corfu server to the network. Docker container: {}", params.getName());
+
+        try {
+            execCommand(IpTablesUtil.cleanInput());
+            execCommand(IpTablesUtil.cleanOutput());
+        } catch (DockerException | InterruptedException e) {
+            throw new NodeException("Can't reconnect container to docker network " + params.getName(), e);
+        }
+    }
+
+    /**
+     * Resume a {@link CorfuServer}
+     *
+     * @throws NodeException this exception will be thrown if the node can not be resumed
+     */
+    @Override
+    public void resume() {
+        log.info("Resuming the corfu server: {}", params.getName());
+
+        try {
+            ContainerInfo container = docker.inspectContainer(params.getName());
+            if (!container.state().paused()) {
+                log.warn("The container `{}` is not paused, should pause before resuming", container.name());
+                return;
+            }
+            docker.unpauseContainer(params.getName());
+        } catch (DockerException | InterruptedException ex) {
+            throw new NodeException("Can't resume container " + params.getName(), ex);
+        }
+    }
+
+    @Override
+    public String getIpAddress() {
+        return ipAddress.get();
+    }
+
+    /**
+     * Deploy and start docker container, expose ports, connect to a network
+     *
+     * @return docker container id
+     */
+    private String deployContainer() {
+        ContainerConfig containerConfig = buildContainerConfig();
+
+        String id;
+        try {
+            ContainerCreation container = docker.createContainer(containerConfig, params.getName());
+            id = container.id();
+
+            addShutdownHook();
+
+            docker.disconnectFromNetwork(id, "bridge");
+            docker.connectToNetwork(id, docker.inspectNetwork(universeParams.getNetworkName()).id());
+
+            docker.startContainer(id);
+
+            ipAddress.set(docker.inspectContainer(id).networkSettings().ipAddress());
+        } catch (InterruptedException | DockerException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)